include/flox/resolver/environment.hh

```diff
@@ -52,86 +64,195 @@ private:
   std::optional<Lockfile> oldLockfile;
 
   /** New/modified lockfile being edited. */
-  Lockfile lockfile;
+  std::optional<LockfileRaw> lockfileRaw;
 
   std::optional<RegistryRaw> combinedRegistryRaw;
 
+  std::optional<Options> combinedOptions;
+
+  std::optional<pkgdb::PkgQueryArgs> combinedBaseQueryArgs;
+
   /** A registry of locked inputs. */
   std::optional<RegistryRaw> lockedRegistry;
 
-  std::shared_ptr<pkgdb::PkgDbInputFactory> dbFactory;
-
   std::shared_ptr<Registry<pkgdb::PkgDbInputFactory>> dbs;
 
+  static LockedPackageRaw
```

Comment:
This name is kind of confusing, I wouldn't expect to see "locked" and "raw" together.

Reply:
Raw indicates that this is _raw_ data being de-serialized.

The underlying data represents a _locked/resolved package_.